kokain-ksp/src/main/java/com/schwarz/kokain/ksp/factory/EBeanModelFactory.kt

```diff
@@ -0,0 +1,36 @@
+package com.schwarz.kokain.ksp.factory
+
+import com.google.devtools.ksp.KspExperimental
+import com.google.devtools.ksp.getAnnotationsByType
+import com.google.devtools.ksp.getVisibility
+import com.google.devtools.ksp.processing.KSPLogger
+import com.google.devtools.ksp.processing.Resolver
+import com.google.devtools.ksp.symbol.KSAnnotated
+import com.google.devtools.ksp.symbol.KSClassDeclaration
+import com.google.devtools.ksp.symbol.Visibility
+import com.schwarz.kokain.api.EBean
+import com.schwarz.kokain.ksp.model.EBeanModel
+import com.schwarz.kokain.ksp.validation.PreValidator
+import com.squareup.kotlinpoet.KModifier
+
+class EBeanModelFactory(val logger: KSPLogger, resolver: Resolver) {
+
+    private val preValidator = PreValidator(logger, resolver)
+
+    @OptIn(KspExperimental::class)
+    fun create(element: KSAnnotated): EBeanModel? {
+        (element as? KSClassDeclaration)?.let {
+            if (preValidator.validateEbean(it)) {
+                val scope = it.getAnnotationsByType(EBean::class).first().scope
+                val simpleName = it.simpleName.asString()
+                val sPackage = it.packageName.asString()
+                val visibility = when (it.getVisibility()) {
+                    Visibility.INTERNAL -> KModifier.INTERNAL
+                    else -> KModifier.PUBLIC
```

Comment:
why don't we support private and protected here?
I guess private does not work because we generate a subclass, but protected should work, shouldn't it?

Reply:
no because where do you want to inject a class which is only visible for the child class?

Reply:
Ok, so the visibility is for the entire class, not just a property?
Yes, in that case, it makes sense 👍 
Sorry, I was not sure whether it was for classes or properties